docs/DeveloperGuide.md

```diff
@@ -234,6 +234,50 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 _{more aspects and alternatives to be added}_
 
+### Tag adding feature
+
+### Implementation
+
+The tag adding mechanism is facilitated bt `TagAddCommand` and `TagAddCommandParser`.
+Additionally, The mechanism utilises the following operations in `UniqueTagList` and `UniquePersonList`.
+
+* `boolean hasTag(Tag tag)` - Checks if the tagList has the tag.
+* `void setPerson(Person target, Person editedPerson)` - Sets the same person with the new tag.
+* 
+
+These operations are exposed in the `Model` interface under the same method name.
+
+Given below is an example usage scenario and how the sorting mechanism behaves at each step.
+
+Step 1. The user executes `tag add 1 t/friend` command to add the tag, `friend`, to the person indicated by the `INDEX`, 1.
+`TagAddCommandParser` calls  `ArgumentTokenizer#tokenizeToList()` to separate the parameters of `1` and `t/friend`.
+
+Step 2. The `tag add` command collects the list of persons shown, containing them in `lastShownList = model#getFilteredPersonList()`.
+
+Step 3. The `tag add` command checks if the index is valid, calling `index.getZeroBased()`.
+
+Step 4. The `tag add` command receives the person indicated by the index, containing it in `personToEdit = lastShownList#get(int index)`.
+
+Step 5. The `tag add` command creates the same person with the new tag included, containing it in `editedPerson = createEditedPerson(Person personToEdit, Tag tag)`.
+
+Step 6. The `tag add` command replaces the old person with the new, updated person, calling `model#setPerson(Person target, Person editedPerson)`.
+
+The following activity diagram summarizes what happens when a user executes a tag add command:
+
+(insert activity diagram here)
+
+#### Design considerations:
```

Comment:
`Design consideration`, remove the `s:` so that this section looks nicer in the TOC